src/components/ResponsiveTable/ResponsiveTable.tsx

```diff
@@ -0,0 +1,56 @@
+import { Theme } from "@material-ui/core/styles";
+import Table from "@material-ui/core/Table";
+import { makeStyles } from "@material-ui/styles";
+import classNames from "classnames";
+import React from "react";
+
+const useStyles = makeStyles(
+  (theme: Theme) => ({
+    [theme.breakpoints.up("sm")]: {
+      "& table": {
+        tableLayout: "fixed"
+      }
+    },
+    root: {
+      "& table": {
+        tableLayout: "auto"
+      },
+      overflowX: "auto",
+      width: "100vw"
+    },
+    rootDialog: {
+      width: "100%"
+    }
+  }),
+  {
+    name: "ResponsiveTable"
+  }
+);
+
+interface ResponsiveTableProps {
+  children: React.ReactNode | React.ReactNodeArray;
+  className?: string;
+  dialog?: boolean;
+  key?: string;
+}
+
+const ResponsiveTable: React.FC<ResponsiveTableProps> = props => {
+  const { children, className, dialog, key } = props;
```

Comment:
I don't think you understand what `key` prop is about:
https://reactjs.org/docs/lists-and-keys.html#keys